cbits/energy-rapl.c

```diff
@@ -1,28 +1,40 @@
 #include <unistd.h>
+#include <pthread.h>
 #include "rapl.h"
 
 static int fd_msr = 0;
 static struct rapl_units r_units;
 static int units_ok;
 
+static double energy_sum = 0;
+pthread_t thread;
+
+void *thread_sum_energy_packet(void *arg)
+{
+    for (;;)
+    {
+        sleep(15);
+        struct rapl_raw_power_counters rpc;
+        rapl_get_raw_power_counters(fd_msr, &r_units, &rpc);
```

Comment:
Consider using [rapl_get_power_diff()](https://github.com/anyc/librapl/blob/5d823570786595b3d527ed97a8e7a57e698adceb/rapl.h#L112) here. You can accumulate just the diff and if an overflow happens (diff < 0) you can discard the current one.